features/netsocket/cellular/generic_modem_driver/OnboardCellularInterface.h

```diff
@@ -18,7 +18,39 @@
 
 #include "CellularContext.h"
 #ifdef CELLULAR_DEVICE
-typedef mbed::CellularContext OnboardCellularInterface;
+using namespace mbed;
+MBED_DEPRECATED_SINCE("mbed-os-5.9", "This API will be deprecated, use CellularBase::get_default_instance() instead.")
+class OnboardCellularInterface : public CellularBase
+{
+public:
+    OnboardCellularInterface() {
+        context = static_cast<CellularContext*>(CellularContext::get_default_instance());
```

Comment:
This cast shouldn't be required - `CellularContext::get_default_instance()` should be returning a CellularContext *`.

(Alternatively this could have `context` be a `CellularBase` pointer, but `CellularContext::get_default_instance()` is definitely wrong).

Reply:
True, changed to return CellularContext* and removed casting as it's not needed anymore.